### ISSUE TITLE
expect.it assertions in combinations with `when called` does not work.

### ISSUE THREAD

**Issue by sunesimonsen** [unexpectedjs/unexpected-sinon#30]
Does not fail:

```js
expect(function () { return 'foo' }, 'when called', expect.it('to be a function'))
```

There might be a general bug with `<assertion?>`, but I don't know.

**Comment by papandreou** [comment]
D'oh, it seems that only these two are supported:

```js
expect(function () { return 'foo' }, 'when called', 'to be a function');
expect(function () { return 'foo' }, 'when called to be a function');
```

It's a bit of an oversight that `expect.it` isn't supported, and especially that it just passes silently.

**Comment by sunesimonsen** [comment]
Yes, if it would just fail :-)

**Comment by papandreou** [comment]
Fixed in Unexpected 10.7.0.
